cabra/src/pages/TaskDetails.tsx

```diff
@@ -1,36 +1,52 @@
 import "twin.macro";
 
-import { useNavigate, useParams } from "react-router-dom";
+import { ArrowLeftIcon, PencilAltIcon } from "@heroicons/react/solid";
+import { Link, useNavigate, useParams } from "react-router-dom";
 
 import { ITask } from "../types/task";
 import Layout from "./components/Layout";
+import NavigationButton from "./components/NavigationButton";
 import Task from "./components/Task";
 import axios from "axios";
+import { routeHelpers } from "../routes";
 import { useQuery } from "react-query";
 
 export default function TaskDetailsPage() {
   const navigate = useNavigate();
-  const { id } = useParams();
+  const { id } = useParams() as { id: string };
```

Comment:
```suggestion
  const { id } = useParams<{ id: string }>();
```

Reply:
That's not gonna work unfortunately 😢 The generic is getting `Partial`ized by `react-router` and `id` still comes out as `string | undefined` while I need it to be just `string`.